src/routes/zome_call/tests/responses.rs

```diff
@@ -1,54 +1,206 @@
+}
+
+#[tokio::test]
+async fn fresh_nonce_error() {
+    let mut app_call = MockAppCall::new();
+    app_call
+        .expect_handle_zome_call()
+        .returning(|_, _, _, _, _| {
+            Box::pin(async move {
+                Err(crate::HcHttpGatewayError::HolochainError(
+                    ConductorApiError::FreshNonceError("nonce_kaputt".into()),
+                ))
+            })
+        });
+    let router = create_test_router(app_call);
+    let (status_code, body) = router
+        .request(&format!("/{DNA_HASH}/{APP_ID}/coordinator/fn_name"))
+        .await;
+    assert_eq!(status_code, StatusCode::INTERNAL_SERVER_ERROR);
+    assert_eq!(body, r#"{"error":"Something went wrong"}"#);
+}
+
+#[tokio::test]
+async fn io_error() {
+    let mut app_call = MockAppCall::new();
+    app_call
+        .expect_handle_zome_call()
+        .returning(|_, _, _, _, _| {
+            Box::pin(async move {
+                Err(crate::HcHttpGatewayError::HolochainError(
+                    ConductorApiError::IoError(std::io::Error::other("ssd not found")),
+                ))
+            })
+        });
+    let router = create_test_router(app_call);
+    let (status_code, body) = router
+        .request(&format!("/{DNA_HASH}/{APP_ID}/coordinator/fn_name"))
+        .await;
+    assert_eq!(status_code, StatusCode::INTERNAL_SERVER_ERROR);
+    assert_eq!(body, r#"{"error":"Something went wrong"}"#);
+}
+
+#[tokio::test]
+async fn sign_zome_call_error() {
+    let mut app_call = MockAppCall::new();
+    app_call
+        .expect_handle_zome_call()
+        .returning(|_, _, _, _, _| {
+            Box::pin(async move {
+                Err(crate::HcHttpGatewayError::HolochainError(
+                    ConductorApiError::SignZomeCallError("unsigned".to_string()),
+                ))
+            })
+        });
+    let router = create_test_router(app_call);
+    let (status_code, body) = router
+        .request(&format!("/{DNA_HASH}/{APP_ID}/coordinator/fn_name"))
+        .await;
+    assert_eq!(status_code, StatusCode::INTERNAL_SERVER_ERROR);
+    assert_eq!(body, r#"{"error":"Something went wrong"}"#);
+}
+
+#[tokio::test]
+async fn websocket_error() {
+    let mut app_call = MockAppCall::new();
+    app_call
+        .expect_handle_zome_call()
+        .returning(|_, _, _, _, _| {
+            Box::pin(async move {
+                Err(crate::HcHttpGatewayError::HolochainError(
+                    ConductorApiError::WebsocketError(
+                        // WebsocketError is not exposed.
+                        std::io::Error::other("websocket closed").into(),
+                    ),
+                ))
+            })
+        });
+    let router = create_test_router(app_call);
+    let (status_code, body) = router
+        .request(&format!("/{DNA_HASH}/{APP_ID}/coordinator/fn_name"))
+        .await;
+    assert_eq!(status_code, StatusCode::INTERNAL_SERVER_ERROR);
+    assert_eq!(body, r#"{"error":"Something went wrong"}"#);
+}
```

Comment:
These tests look correct for these error types. How about errors from Holochain that indicate a failure within the zome call that should go back to the user? If possible we want to make this debugable by the end user.

"JSON error response with an error field with a hard-coded string for conductor errors or the zome error message if this was an error raised by the target hApp."

Reply:
Yes, all we have here is what the AppWebsocket returns, and that's a `holochain_client::ConductorApiError`. That has to be modified then.

Reply:
Right, right, we talked about this and it's all strings for now. Okay fine with me. 

We might find that we get questions about this and we need to look at upping the log level for failed zomes so there's a way in for debugging.